packages/api/src/feature_flag/service/create.ts

```diff
@@ -0,0 +1,50 @@
+import { logger } from "@stela/logger";
+import createError from "http-errors";
+import type { CreateFeatureFlagRequest, FeatureFlagRow } from "../models";
+import { db } from "../../database";
+
+export const createFeatureFlag = async (
+  featureFlagData: CreateFeatureFlagRequest
+): Promise<FeatureFlagRow> => {
+  const featureFlagsWithName = async (name: string): Promise<boolean> => {
+    const result = await db
+      .sql<FeatureFlagRow>("feature_flag.queries.get_feature_flag", {
+        name,
+      })
+      .catch((err) => {
+        logger.error(err);
+        throw new createError.InternalServerError(
+          "Failed to retrieve feature flags"
+        );
+      });
+
+    return result.rows.length > 0;
+  };
+
+  if (await featureFlagsWithName(featureFlagData.name)) {
+    throw new createError.BadRequest("Feature flag already exists");
+  }
```

Comment:
Rather than making an extra call to the database to see whether a feature flag of the same name already exists, I'd recommend adding a unique constraint to `feature_flag.name` in the database (docs on how we do migrations: https://github.com/PermanentOrg/back-end?tab=readme-ov-file#database-migration). Then you can check for an error related to that constraint (similar to what we do here: https://github.com/PermanentOrg/back-end?tab=readme-ov-file#database-migration) on the insert operation.